assets/src/hooks/useNearestIntersection.ts

```diff
@@ -1,38 +1,46 @@
-import { useEffect, useState } from "react"
+import { useCallback } from "react"
 import { fetchNearestIntersection } from "../api"
-import { FetchResult, loading } from "../util/fetchResult"
+import { FetchResult, loading, ok, fetchError } from "../util/fetchResult"
+import { useApiCall } from "./useApiCall"
 
+/**
+ * @deprecated use {@linkcode useNearestIntersection2}
+ */
 export const useNearestIntersection = (
   latitude: number | undefined,
   longitude: number | undefined
-): FetchResult<string | null> => {
-  const [result, setResult] = useState<FetchResult<string | null>>({
-    is_loading: true,
-  })
-
-  useEffect(() => {
-    let shouldUpdate = true
-
-    setResult((oldResult) => {
-      return { ...oldResult, is_loading: true }
-    })
+): FetchResult<string> => {
+  const result = useNearestIntersection2({ latitude, longitude })
 
-    if (latitude && longitude) {
-      fetchNearestIntersection(latitude, longitude).then((result) => {
-        if (shouldUpdate) {
-          if (result) {
-            setResult({ ok: result })
-          } else {
-            setResult({ is_error: true })
-          }
-        }
-      })
-    } else setResult(loading())
-
-    return () => {
-      shouldUpdate = false
+  // Temporary code to convert to the old function shape
+  if (result.isLoading) {
+    if (result.result !== undefined) {
+      return { ...ok(result.result), ...loading() }
+    } else {
+      return loading()
+    }
+  } else {
+    if (result.result === undefined || result.result === null) {
+      return fetchError()
     }
-  }, [latitude, longitude])
 
-  return result
+    return ok(result.result)
+  }
 }
+
+const useNearestIntersection2 = ({
```

Comment:
Would love a better name

Reply:
Could this just be in-lined into `useNearestIntersection`?

Reply:
I want `useNearestIntersection` deprecated and to move off of the `FetchResult` types at some point in the future. I should document that in the original function

Reply:
Yep!
I created a ticket for this so we can do it in the future and added it to the description for tracking purposes https://github.com/mbta/skate/compare/4b7bdb3144c9ded68ca466ea5149c00a24ba8fae..f24cf71f23e5f32978fcc614dfac47424856ea7e

Reply:
I actually like having a silly name like `useNearestIntersection2` - it falls into the category of [making wrong code look wrong](https://www.joelonsoftware.com/2005/05/11/making-wrong-code-look-wrong/), or in this case, making deprecated code look deprecated,

A possible other approach would be to rename the existing `useNearestIntersection` to `useNearestIntersectionDeprecated`, and then introduce the current `useNearestIntersection2` as `useNearestIntersection`. That way the "bad" function is the one with the silly name, rather than the "good" function.

Reply:
(By the way, most of the specific examples in the article linked are things that can and should be fixed by either a linter or a type-checker, but the general idea of making bad things look bad stands, I think)